src/payment/payment-action-creator.ts

```diff
@@ -37,10 +37,18 @@ export default class PaymentActionCreator {
 
     initializeOffsitePayment(
         methodId: string,
-        gatewayId?: string
+        gatewayId?: string,
+        instrumentId?: string,
+        shouldSaveInstrument?: boolean
     ): ThunkAction<InitializeOffsitePaymentAction, InternalCheckoutSelectors> {
         return store => {
-            const payload = this._paymentRequestTransformer.transform({ gatewayId, methodId }, store.getState());
+            const paymentData = {
+                formattedPayload: {
+                    bigpay_token: instrumentId ? instrumentId : null,
```

Comment:
I think you only ever want to send one or the other - not both, because it doesn't make sense for you to indicate that you want to vault an instrument when you're already paying with a vaulted instrument. The constraint could either be applied on the interface or implementation level. i.e.:

```ts
let paymentData: FormattedPayload<FormattedHostedInstrument | FormattedVaultedInstrument> | undefined;

if (instrumentId) {
    paymentData = { bigpay_token: instrumentId };
} else if (shouldSaveInstrument) {
    paymentData = { vault_payment_instrument: shouldSaveInstrument };
}
```